exporter/elasticsearchexporter/internal/serializer/otelserializer/otelserializer.go

```diff
@@ -3,9 +3,35 @@
 
 package otelserializer // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter/internal/serializer/otelserializer"
 
-type Serializer struct{}
+import (
+	"sync"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter/internal/lru"
+)
+
+const (
+	knownExecutablesCacheSize             = 16 * 1024
+	knownFramesCacheSize                  = 128 * 1024
+	knownTracesCacheSize                  = 128 * 1024
+	knownUnsymbolizedFramesCacheSize      = 128 * 1024
+	knownUnsymbolizedExecutablesCacheSize = 16 * 1024
+
+	minILMRolloverTime = 3 * time.Hour
+)
+
+type Serializer struct {
+	mu sync.Mutex
```

Comment:
using sync.Mutex to avoid creating a profiling specific struct is good, but sync.Once will be better as there will be a fast path after the initialization is done